Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,209 @@
+                    PKLog.debug("ended event: \(event)")
+                    
+                    if (!strongSelf.playReached100Percent) {
+                        strongSelf.playReached100Percent = true
+                        strongSelf.sendAnalyticsEvent(action: KavaEventType.playReached100Percent)
+                    }
+                })
+                
+            case let e where e.self == PlayerEvent.playbackInfo:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("playbackInfo event: \(event)")
+                    strongSelf.indicatedBitrate = event.playbackInfo?.indicatedBitrate
+                })
+            
+            case let e where e.self == PlayerEvent.trackChanged:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("trackChanged event: \(event)")
+                    if let track = event.selectedTrack {
+                        if (track.title.contains("event.selectedTrack?.title")) {
+                            strongSelf.currentCaptionLanguage = event.selectedTrack?.language
+                            strongSelf.sendAnalyticsEvent(action: KavaEventType.captions)
+                        }
+                    }
+                })
+                
+            case let e where e.self == PlayerEvent.error:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("PlayerEvent error event: \(event)")
+                    strongSelf.errorCode = (event.error?.code)!
+                    strongSelf.sendAnalyticsEvent(action: KavaEventType.error)
+                })
+            default: assertionFailure("all events must be handled")
+            }
+        }
+        
+        self.messageBus?.addObserver(self, events: [AdEvent.error], block: { [weak self] (event) in
+            guard let strongSelf = self else { return }
+            PKLog.debug("AdEvent error event: \(event)")
+            strongSelf.errorCode = (event.error?.code)!
+            strongSelf.sendAnalyticsEvent(action: KavaEventType.error)
+        })
+    }
+    
+    /************************************************************/
+    // MARK: Private Implementation
+    /************************************************************/
+    
+    private func handleStateChanged(event: PKEvent) {
+        switch event.newState {
+        case .idle:
+            PKLog.debug("state changed to idle")
+        case .ended:
+            PKLog.info("media ended")
+        case .ready:
+            if self.isBuffering {
+                self.isBuffering = false
+            }
+            
+            self.sendMediaLoaded()
```

Comment:
you send media loaded in ready and in loadedMetadata event observer also, should be send once no?